src/main/java/com/springweb/dv_spring_web_mongo/service/ProjectService.java

```diff
@@ -26,25 +23,33 @@
         for (Project project : list) {
             listWithDto.add(project.convertToDTO());
         }
-        return listWithDto;
+        if (listWithDto.isEmpty()) throw new ProjectNotFoundException("Database of projects is empty");
+        else return listWithDto;
     }
 
     public ProjectDTO getProjectById(String id) {
-        Optional<Project> project = projectRepository.findById(id);
-        return project.get().convertToDTO();
+        Optional<Project> optional = projectRepository.findById(id);
+        if(optional.isEmpty()) throw new ProjectNotFoundException("Project with id '" + id + "' not found");
+        else return optional.get().convertToDTO();
     }
 
     public void addNewProject(ProjectDTO projectDTO) {
         projectRepository.save(projectDTO.convertToProject());
     }
 
     public void changeProjectName(ProjectDTO projectDTO, String id) {
-        Project project = projectRepository.findById(id).get();
-        project.setProjectName(projectDTO.getProjectName());
-        projectRepository.save(project);
+        Optional<Project> optional = projectRepository.findById(id);
```

Comment:
the code is repeated several times:
```java
Optional<Project> optional = projectRepository.findById(id);
        if (optional.isEmpty()) throw new ProjectNotFoundException("Project with id '" + id + "' not found");
```
can you create new method with signature `private Project getById(String projectId)` and use?